Stop using some deprecated methods in WebKitLegacy

brentfulgham opened the pull request:
#### 56def202fc3c2585e0be1c8eeecbb5cfd408e1a8
<pre>
Stop using some deprecated methods in WebKitLegacy
<a href="https://bugs.webkit.org/show_bug.cgi?id=293929">https://bugs.webkit.org/show_bug.cgi?id=293929</a>
<a href="https://rdar.apple.com/152461747">rdar://152461747</a>

Reviewed by Sam Weinig.

This patch builds on Bug 293059, and switches from deprecated AppKit methods to the
modern equivalents. There is no change in behavior.

* Source/WebKitLegacy/mac/WebInspector/WebNodeHighlight.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _updateBackgroundAndUpdatesWhileOffscreen]):
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
(-[WebFullScreenController enterFullScreen:willEnterFullscreen:didEnterFullscreen:]):
(-[WebFullScreenController _startEnterFullScreenAnimationWithDuration:]):
(-[WebFullScreenController _startExitFullScreenAnimationWithDuration:]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView _updateMouseoverWithFakeEvent]):
(-[WebHTMLView _autoscroll]):
(-[WebHTMLView accessibilityHitTest:]):
(-[WebHTMLView _colorAsString:]):
* Source/WebKitLegacy/mac/WebView/WebTextCompletionController.mm:
(-[WebTextCompletionController _buildUI]):
(-[WebTextCompletionController _placePopupWindow:]):

Canonical link: <a href="https://commits.webkit.org/295827@main">https://commits.webkit.org/295827@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e832d81ce5b2cec5bed81ab94939fe3ac96024e3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106098 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25847 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16241 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111295 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56694 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26503 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34350 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80590 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109102 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20942 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95746 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60912 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20517 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13844 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56132 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90309 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13879 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114151 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33236 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/24636 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89672 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33600 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91974 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89361 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34225 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12041 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28809 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17220 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33161 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38573 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32907 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36257 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34505 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebKitLegacy/mac/Misc/WebKitErrors.m:
```diff
@@ -71,13 +71,11 @@ - (instancetype)_webkit_initWithDomain:(NSString *)domain code:(int)code URL:(NS
     NSDictionary *userInfo = [NSDictionary dictionaryWithObjectsAndKeys:
         URL, @"NSErrorFailingURLKey",
-        [URL absoluteString], NSURLErrorFailingURLStringErrorKey,
+        [URL absoluteString], NSURLErrorFailingURLErrorKey,
```
I think if you use NSURLErrorFailingURLErrorKey the value is supposed to an NSURL (https://developer.apple.com/documentation/foundation/nsurlerrorfailingurlerrorkey#Discussion)

brentfulgham replied:
Doh! You are correct. I'll revise.

brentfulgham replied:
I'm just going to remove this change entirely, just in case some software is looking for that deprecated key to consume the error.

brentfulgham committed:
Stop using some deprecated methods in WebKitLegacy
https://bugs.webkit.org/show_bug.cgi?id=293929
rdar://152461747

Reviewed by Sam Weinig.

This patch builds on Bug 293059, and switches from deprecated AppKit methods to the
modern equivalents. There is no change in behavior.

* Source/WebKitLegacy/mac/WebInspector/WebNodeHighlight.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _updateBackgroundAndUpdatesWhileOffscreen]):
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
(-[WebFullScreenController enterFullScreen:willEnterFullscreen:didEnterFullscreen:]):
(-[WebFullScreenController _startEnterFullScreenAnimationWithDuration:]):
(-[WebFullScreenController _startExitFullScreenAnimationWithDuration:]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView _updateMouseoverWithFakeEvent]):
(-[WebHTMLView _autoscroll]):
(-[WebHTMLView accessibilityHitTest:]):
(-[WebHTMLView _colorAsString:]):
* Source/WebKitLegacy/mac/WebView/WebTextCompletionController.mm:
(-[WebTextCompletionController _buildUI]):
(-[WebTextCompletionController _placePopupWindow:]):

Canonical link: https://commits.webkit.org/295827@main
